Library/Homebrew/language/node.rb

```diff
@@ -0,0 +1,32 @@
+module Language
+  module Node
+    def self.npm_cache_config
+      "cache=#{HOMEBREW_CACHE}/npm_cache\n"
+    end
+
+    def self.setup_npm_environment
+      # explicitly use our npm and node-gyp executables instead of the user
+      # managed ones in HOMEBREW_PREFIX/lib/node_modules which might be broken
+      ENV.prepend_path "PATH", "#{Formula["node"].opt_libexec}/npm/bin"
+      # explicitly set npm's cache path to HOMEBREW_CACHE/npm_cache to fix
+      # issues caused by overriding $HOME (long build times, high disk usage)
+      # https://github.com/Homebrew/brew/pull/37#issuecomment-208840366
+      npmrc = Pathname.new("#{ENV["HOME"]}/.npmrc")
+      npmrc.write npm_cache_config unless npmrc.exist?
+    end
+
+    def self.std_npm_install_args(libexec)
+      setup_npm_environment
+      # tell npm to not install .brew_home by adding it to .npmignore
+      open('.npmignore', 'a') { |f| f.write( "\n.brew_home\n") }
```

Comment:
I will changed the quotes (silly copy and paste mistake).

@UniqMartin The npm command wouldn't work if we aren't in the right directory, so I think we can assume this here.
There is also no guarantee that a `.npmignore` file already exists. I'm relaying on the 'append if the file exists and create a new one if not'-functionality here.